clients/vault/src/execution.rs

```diff
@@ -259,6 +259,42 @@ impl Request {
 	}
 }
 
+/// executes open request based on the transaction
+async fn _execute_open_requests(
+	transaction: TransactionResponse,
+	oracle_agent: Arc<OracleAgent>,
+	parachain_rpc: SpacewalkParachain,
+	request: Request,
+) {
+	// max of 3 retries for failed request execution
+	let max_retries = 3;
+	let mut retry_count = 0;
+
+	if let Ok(tx_env) = transaction.to_envelope() {
+		let slot = transaction.ledger as Slot;
+		while retry_count < max_retries {
+			match oracle_agent.get_proof(slot).await {
+				Ok(proof) => {
+					if let Err(e) =
+						request.execute(parachain_rpc.clone(), tx_env.clone(), proof).await
+					{
+						tracing::error!("Failed to execute request #{}: {}", request.hash, e);
+						retry_count += 1; // increase retry count
+						continue
+					}
+					break
```

Comment:
Can we rewrite it with a match statement to make it more obvious that we break the while function on successful submit? It took me a second to figure out why we `continue` in one case and `break` afterwards. Maybe that makes it clearer.